src/activity_actor.cpp

```diff
@@ -409,12 +412,71 @@ std::unique_ptr<activity_actor> open_gate_activity_actor::deserialize( JsonIn &j
     return actor.clone();
 }
 
+void consume_activity_actor::start( player_activity &act, Character &who )
+{
+    item &comest = who.get_consumable_from( *loc.get_item() );
+
+    const int charges = std::max( comest.charges, 1 );
+    int volume = units::to_milliliter( comest.volume() ) / charges;
```

Comment:
comest.volume() is giving you the volume of the whole stack, don't you just want the volume of one from here?

Reply:
I'm copying the logic used for displaying the amount of food you are eating in the consume menu.